source/source.go

```diff
@@ -97,21 +131,19 @@ func (s *Source) Open(ctx context.Context, position opencdc.Position) error {
 func (s *Source) Read(ctx context.Context) (opencdc.Record, error) {
 	sdk.Logger(ctx).Debug().Msg("Reading a record from Amazon Redshift Source...")
 
-	hasNext, err := s.iterator.HasNext(ctx)
-	if err != nil {
-		return opencdc.Record{}, fmt.Errorf("has next: %w", err)
-	}
-
-	if !hasNext {
-		return opencdc.Record{}, sdk.ErrBackoffRetry
+	if s == nil || s.ch == nil {
+		return opencdc.Record{}, fmt.Errorf("error source not opened for reading")
```

Comment:
Nitpick: we can simplify the error: `errors.New("source not opened for reading")`.